pkg/util/migrations/migrations.go

```diff
@@ -112,3 +111,21 @@ func VMIMigratableOnEviction(clusterConfig *virtconfig.ClusterConfig, vmi *v1.Vi
 	}
 	return false
 }
+
+func InterruptedMigrationForVMI(migrationIndexer cache.Indexer, vmi *v1.VirtualMachineInstance) (*v1.VirtualMachineInstanceMigration, error) {
+	objs, err := migrationIndexer.ByIndex(cache.NamespaceIndex, vmi.Namespace)
+	if err != nil {
+		return nil, err
+	}
+	for _, obj := range objs {
+		migration := obj.(*v1.VirtualMachineInstanceMigration)
+		if migration.Spec.VMIName == vmi.Name {
+			if migration.Status.Phase != v1.MigrationInterrupted {
+				log.Log.Warningf("Found an unexpected migration for VMI %s/%s: %s", vmi.Namespace, vmi.Name, migration.Name)
```

Comment:
I don't think this is correct. We could have cases where we have a failed migration, due to some connection issues for example, and then we will try again (in case of the workload updated), and then the migration could be in an interrupted state.

I would loop over all the migrations and then check that there is only one in Interrupted state.

Reply:
I have a hard time seeing how there could be multiple migrations in interrupted state, but what do you suggest we do in that case?

Reply:
You are probably right. I was thinking if there is some clean-up problem and we have some migration object left, and then we re trigger a migration which is again interrupted. But when a migration is interrupted the VM shoudn't be running, and then we check the migration at restart. So, it should be safe.